### PR TITLE
OpInfo: ravel, view, view_as

### PR DESCRIPTION
**Pull request by kshitij12345**
Reference: #54261

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -4946,6 +4992,35 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
...
+           dtypesIfCUDA=all_types_and_complex_and(torch.bool, torch.float16, torch.bfloat16),
+           dtypesIfROCM=all_types_and_complex_and(torch.bool, torch.float16, torch.bfloat16),
+           supports_out=False,
+           skips=(
+               # Because view does not have a function variant.
+               SkipInfo('TestCommon', 'test_variant_consistency_jit'),),
+           sample_inputs_func=sample_inputs_view,
+           ),
+    OpInfo('view_as',
+           op=lambda x, other: x.view_as(other),
```
Comment (kshitij12345) [inline torch/testing/_internal/common_methods_invocations.py:5014]: Add function variant?

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -4946,6 +4992,35 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
...
                    sample_kwargs=lambda device, dtype, input: ({'n': 4}, {'n': 4})),
+    OpInfo('ravel',
+           dtypesIfCPU=all_types_and_complex_and(torch.bool, torch.float16, torch.bfloat16),
+           dtypesIfCUDA=all_types_and_complex_and(torch.bool, torch.float16, torch.bfloat16),
+           dtypesIfROCM=all_types_and_complex_and(torch.bool, torch.float16, torch.bfloat16),
+           supports_out=False,
+           sample_inputs_func=sample_inputs_ravel,
+           ),
+    OpInfo('view',
+           op=lambda x, shape: x.view(shape),
```
Comment (kshitij12345) [inline torch/testing/_internal/common_methods_invocations.py:5003]: Add function variant?

Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:5003]: I think there were some discussions on this topic but we didn't follow-through. Let's just leave it without a function variant for the moment

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -4946,6 +4992,35 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
...
+           ),
+    OpInfo('view_as',
+           op=lambda x, other: x.view_as(other),
+           dtypesIfCPU=all_types_and_complex_and(torch.bool, torch.float16, torch.bfloat16),
+           dtypesIfCUDA=all_types_and_complex_and(torch.bool, torch.float16, torch.bfloat16),
+           dtypesIfROCM=all_types_and_complex_and(torch.bool, torch.float16, torch.bfloat16),
+           supports_out=False,
+           skips=(
+               # Because view_as does not have a function variant.
+               SkipInfo('TestCommon', 'test_variant_consistency_jit'),),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:5021]: An interesting follow-up opportunity is revising test_variant_consistency_jit to no longer assume the op has a function variant

File: torch/testing/_internal/common_methods_invocations.py [hunk 4]
```diff
@@ -2970,6 +2970,52 @@ def _gather(shape, index_dim, max_indices):
...
+    cases = (((S, S, S), (S * S, S)),
+             ((), ()),
+             ((), (1, 1)),
+             )
+
+    def generator():
+        for case in cases:
+            shape, shape_other = case
+            yield(SampleInput(make_arg(shape, requires_grad=requires_grad),
+                              args=(make_arg(shape_other, requires_grad=False), )))
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:3014]: Nice nuance

**Comment by mruberry** [comment]
Note: blocked on the index_fill issue.

Other than the blocking issue this looks great, @kshitij12345! Ping me once the tests are passing

**Comment by kshitij12345** [comment]
@mruberry PR is ready :)

**Review by mruberry — APPROVED** [review]
Nice work as usual, @kshitij12345! We'll kill method_tests() yet!

**Comment by mruberry** [comment]
Unfortunately it looks like this picked up a conflict with another OpInfo PR.

We'd hit fewer conflicts in test_fx* if the lists had one element per line and were alphabetical

**Pull request closed without merging**